### PR TITLE
Add SUSEConnect command line register function

### PR DESCRIPTION
**Pull request by Jiawei517**
Add command_register and get_addon_fullname function.
command_register use to register product or modules, get_addon_fullname use to get product full name by abbreviation.
Update deregister by using command_register and get_addon_fullname function

Verification run http://10.67.17.147/tests/1541#step/deregister/

### CODE REVIEW

**Comment by Soulofdestiny** [comment]
Please fix typo ´/regsiter/register/´ in commit message and PR - besides that, LGTM

**Comment by okurz** [comment]
So is this another list of all addons? We have quite some already,  don't we? 

I didn't check other uses of SUSEConnect but I think we have more. Can you check you are not doing anything duplicate?

**Comment by Jiawei517** [comment]
@okurz I had submit this code as a part of zdup migration with module support, but zdup migration is not for SLES15, so I closed that one and open this new one. I'm sure this one is not duplicate. The full addon name list for suseconnect is different.

**Comment by okurz** [comment]
s/suseconnet/SUSEConnect/

**Comment by okurz** [comment]
s/SUSEConnet/SUSEConne**c**t/

File: tests/feature/feature_console/deregister.pm [hunk 1]
```diff
@@ -30,8 +32,8 @@ sub run {
     }
 
     # check if a certain module is registered, e.g. toolchain
-    my $module  = "sle-module-toolchain";
     my $abbrv   = "tcm";
+    my $module  = get_addon_fullname("$abbrv");
```
Comment (okurz) [inline tests/feature/feature_console/deregister.pm]: nice change, appreciated

File: lib/utils.pm [hunk 2]
```diff
@@ -1385,6 +1386,27 @@ sub get_x11_console_tty {
...
+        'asmm'  => 'sle-module-adv-systems-management',
+        'contm' => 'sle-module-containers',
+        'hpcm'  => 'sle-module-hpc',
+        'lgm'   => 'sle-module-legacy',
+        'pcm'   => 'sle-module-public-cloud',
+        'tcm'   => 'sle-module-toolchain',
+        'wsm'   => 'sle-module-web-scripting',
+    );
+    return $product_list{"$addon"};
+}
```
Comment (okurz) [inline lib/utils.pm]: lib/utils.pm is already getting quite big so we should look for a better place. And as this is clearly related to SLE only I recommend a class in "lib". E.g. there is "lib/registration" already that seems quite fitting. WDYT?

Comment (Jiawei517) [reply lib/utils.pm]: I can't agree with you more, lib/utils is over big. Moved it to lib/registration

**Comment by okurz** [comment]
checks failed, trivial whitespace fix with tidy required. But in general I am not so sure about this PR still. IMHO most of the features you add are not used at all.

**Comment by Jiawei517** [comment]
@okurz This SUSEConnet function was used for zdup migration. In openqa zdup migration case, we don't need register the product, it is fine if only migrated the base product. But if we add some module, the zdup migration will failed because cannot found module repo without registered. That's why we need register in zdup migration.
And also from the our PRD, I found the standard process for zdup migration is use the SUSEConnnet tools before migration. I think it should be useful if we don't abandon zdup migration.

**Comment by okurz** [comment]
s/fuction/function/ in the commt message.

@yxususe please review and compare to what you did with suseconnect recently. Is this duplication?

**Comment by yxususe** [comment]
For the part of registering it is basically the same, however the code here does much more with zypper. Besides that this PR also updated deregister module. So one can say that there is a bit code duplication if we don't consider the test scenarios that are in use. I guess I can use some of the functions in this PR.

**Pull request merged**